procedures/unit-testing-comparators.ipf

```diff
@@ -489,7 +489,12 @@ static Function TEST_WAVE_WRAPPER(wv, majorType, flags, [minorType])
 	if(type2 > 0 && type2 <= 2)
 		type = type | 2^(type2 + 3)
 	endif
-	result = (type & majorType) == majorType
+
+	if(majorType == NULL_WAVE)
+		result = type == majorType
+	else
+		result = (type & majorType) == majorType
+	endif
```

Comment:
Thanks. The fix does solve the issue but I'm wondering if we should not also try to fix some more related issues:
- NULL_WAVE can be combined with other bits, but that makes no sense
- NULL_WAVE can be fed into the minorType as well

The issue with bitmasks and 0 (the current value of NULL_WAVE) is known. Can we solve that by using a different non-null value?

Reply:
`NULL_WAVE` could of course get its own non-null value but wouldn't it be simpler if we delete that comparison completely here and use `REQUIRE(!WaveExists())` instead?

Reply:
We currently use `REQUIRE(!WaveExists)` in production but that is IMHO not very nice and quite unidiomatic. It is also too different compared to `REQUIRE_WAVE(wv, NORMAL_WAVE)`. Or am I totally on the wrong track?

Reply:
So we need to
* change NULL_WAVE to NOT = 0x00
* do this only for maior type. To avoid clashes with upstream, I recommend a high enough number here in case WM adds anything here. Not sure which number, though.
* collapse everything to NULL_WAVE if the NULL_WAVE bit was set.

Reply:
I moved the type = NULL_WAVE down and added a check for WaveExists() if both WaveType() functions returned a non-zero value.

NULL_WAVE is now 256

Reply:
@ukos-git Thanks. will merge.